base/tuple.jl

```diff
@@ -20,7 +20,7 @@ endof(t::Tuple) = length(t)
 size(t::Tuple, d) = d==1 ? length(t) : throw(ArgumentError("invalid tuple dimension $d"))
 getindex(t::Tuple, i::Int) = getfield(t, i)
 getindex(t::Tuple, i::Real) = getfield(t, convert(Int, i))
-getindex(t::Tuple, r::AbstractArray{<:Any,1}) = tuple([t[ri] for ri in r]...)
+getindex(t::Tuple, r::AbstractArray{<:Any,1}) = (n = length(eachindex(r)); ntuple(i->t[r[i]], n))
```

Comment:
> Could indexing r with indices 1:n be an issue?

I feel like we are slowly moving to support non traditional indexing, to support things like, e.g., `OffsetArrays`. That's why I wrote it like this.

> Does using ntuple here not potentially hurt inference?

This is already non-inferable so it doesn't hurt changing it to make it faster.